### PR TITLE
test: 💍 setting up cypress

### PR DESCRIPTION
**Pull request by Loonz806**
But ideally this branch would be something of an example of getting e2e setup on a project. Integrating it to a CI/CD pipeline would be the next discussion and heck could probably do that in this project as well on a followup PR

Cool so you can run `yarn build`, then the `yarn e2e:test` for the CI version or run one port with the application and run cy:open on another terminal for writing e2e tests

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "baseUrl": "http://localhost:9000",
+  "integrationFolder": "cypress/e2e",
+  "viewportWidth": 1440,
+  "viewportHeight": 768,
+  "video": false
+}
```
Comment (Loonz806) [inline cypress.json:7]: This file is created to customize the port, and GUI for Cypress. It also can point to different testing directories... so if you wanted to run for certain spaces you can rename the integration folder. I like e2e over integration

File: cypress/fixtures/example.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
```
Comment (Loonz806) [inline cypress/fixtures/example.json:5]: Fixtures are mocked data that you can setup in e2e testing that represents common data bits, think PageObjects if you are from that space

File: cypress/integration/1-getting-started/todo.spec.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
+/// <reference types="cypress" />
+
```
Comment (Loonz806) [inline cypress/integration/1-getting-started/todo.spec.js:2]: Cypress scaffolds out many examples of test writing in the integration folder so you can follow along in how to arrange/act/asset

File: cypress/plugins/index.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+// the project's config changing)
+
+/**
+ * @type {Cypress.PluginConfig}
+ */
+// eslint-disable-next-line no-unused-vars
+module.exports = (on, config) => {
+  // `on` is used to hook into various events Cypress emits
+  // `config` is the resolved Cypress config
+}
```
Comment (Loonz806) [inline cypress/plugins/index.js:22]: This is where you can add plugins from the Cypress plugin space/repo for additional add-ons from the open-source community. https://docs.cypress.io/plugins/directory in the demo I added cypress-axe

File: cypress/support/index.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import "@testing-library/cypress/add-commands";
+import './commands'
+
+// Alternatively you can use CommonJS syntax:
+// require('./commands')
```
Comment (Loonz806) [inline cypress/support/index.js:21]: This is where I added @testing-library which gives the ability to use @testing-library type syntax similar to how we write unit tests. https://github.com/testing-library/cypress-testing-library/tree/main/types/index.d.ts 
Aka DOM Testing Library

File: jsconfig.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "include": [
+    "node_modules/cypress",
+    "./cypress/**/*.js"
+  ]
+}
```
Comment (Loonz806) [inline jsconfig.json:6]: Lots of people who use VS Code get Intellisense for cypress with this addition

File: package.json [hunk 7]
```diff
@@ -6,7 +6,10 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
-    "test": "jest"
+    "test": "jest",
+    "cy:run": "npx cypress run",
+    "cy:open": "npx cypress open",
+    "e2e:test": "npx browserslist@latest --update-db && start-server-and-test start http://localhost:3000 cy:run"
```
Comment (Loonz806) [inline package.json:12]: Runs the e2e suite in one-go, starts a server off the build directory and asserts e2e functionality, hence why start-server-and-test is added, and also I usually update the browserlist in the same task

File: package.json [hunk 8]
```diff
@@ -16,10 +19,13 @@
...
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^12.0.0",
     "babel-jest": "^27.0.6",
+    "cypress": "^7.7.0",
     "jest": "^27.0.6",
-    "msw": "^0.32.0"
+    "msw": "^0.32.0",
+    "start-server-and-test": "^1.12.6"
   }
 }
```
Comment (Loonz806) [inline package.json:31]: Sweet installs

Comment (Loonz806) [inline package.json:31]: Sweet installs to make it dance

**Review by alexUXUI — APPROVED** [review]

File: cypress/.eslintrc.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+  plugins: [
+    "eslint-plugin-cypress"
+  ],
+  extends: [
+    "plugin:cypress/recommended"
+  ],
+  env: {
+    "cypress/globals": true
+  },
+};
```
Comment (Loonz806) [inline cypress/.eslintrc.js:12]: adding a separate linting file for cypress in it's directory or not you can define at root in the eslint file as well

File: cypress/e2e/solarflares.test.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    cy.get('h1').should('exist')
+    cy.findByText('Solar Flares').should('exist');
+  });
+
+  it('should intercept an API call', () => {
+    cy.intercept({
+      method: 'GET',
+    })
+  });
+})
```
Comment (Loonz806) [inline cypress/e2e/solarflares.test.js:15]: Simple test that the page renders, header is present and a GET request is done.. for more on assertions you can check this URL out https://docs.cypress.io/guides/guides/network-requests#Testing-Strategies

File: cypress/fixtures/example.json (hunk 2 above)
Comment (Loonz806) [reply cypress/fixtures/example.json:5]: Or alias to methods on APIS

File: package.json [hunk 11]
```diff
@@ -16,10 +19,14 @@
     "react-query": "^3.18.1"
   },
   "devDependencies": {
+    "@testing-library/cypress": "^7.0.6",
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^12.0.0",
     "babel-jest": "^27.0.6",
+    "cypress": "^7.7.0",
+    "eslint-plugin-cypress": "^2.11.3",
```
Comment (Loonz806) [inline package.json:27]: Forgot about this cool plugin for eslint as well

**Pull request merged**